### PR TITLE
profile_cubit_finished

### PR DESCRIPTION
**Pull request by aleke88884**
Profile screen turned into cubit

### CODE REVIEW

File: lib/features/authentication/ui/screen/password_page.dart [hunk 1]
```diff
@@ -62,8 +63,8 @@ class _PasswordRecoveryPageState extends State<PasswordRecoveryPage> {
             const SizedBox(height: 32.0),
             ElevatedButton(
               onPressed: () {
-                String newPassword = _newPasswordController.text;
-                String confirmPassword = _confirmPasswordController.text;
+                // String newPassword = _newPasswordController.text;
```
Comment (rzhaxibayev) [inline lib/features/authentication/ui/screen/password_page.dart]: remove

File: lib/features/profile/ui/cubit/profile_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_learn_app/features/profile/ui/cubit/profile_state.dart';
+
+class ProfileCubit extends Cubit<ProfileState> {
+  ProfileCubit() : super(const ProfileState());
+
+  void saveData(String name, String phoneNumber) {
+    emit(state.copyWith(name: name, isEdited: true));
```
Comment (rzhaxibayev) [inline lib/features/profile/ui/cubit/profile_cubit.dart]: put in one emit

File: lib/features/profile/ui/screen/profile_screen.dart [hunk 3]
```diff
@@ -71,7 +70,9 @@ class _ProfileScreenState extends State<ProfileScreen> {
                 onPressed: () {
                   Navigator.push(
```
Comment (rzhaxibayev) [inline lib/features/profile/ui/screen/profile_screen.dart]: push named

File: lib/features/profile/ui/screen/profile_settings.dart [hunk 4]
```diff
@@ -33,7 +32,7 @@ class ProfileSettings extends StatelessWidget {
             onPressed: () {
               Navigator.push(
```
Comment (rzhaxibayev) [inline lib/features/profile/ui/screen/profile_settings.dart]: same

**Review by rzhaxibayev — COMMENTED** [review]
Fix all comments

File: lib/main.dart [hunk 5]
```diff
@@ -30,12 +31,17 @@ class MyApp extends StatelessWidget {
...
-        BlocProvider(create: (context) => _di.get<ProfileCubit>()),
-      ], child: const MyHomePage()),
+      initialRoute: AppRouting.home,
+      onGenerateRoute: AppRouting.generateRoute,
+      home: MultiBlocProvider(
+        providers: [
+          BlocProvider(create: (context) => _di.get<CoursesCubit>()),
+          BlocProvider(create: (context) => _di.get<AuthCubit>()),
+          BlocProvider(create: (context) => _di.get<TheoryCubit>()),
+          BlocProvider(create: (context) => _di.get<ProfileCubit>()),
```
Comment (rzhaxibayev) [inline lib/main.dart]: should remove from here as you provide it in app routing

Comment (aleke88884) [reply lib/main.dart]: ok

**Review by rzhaxibayev — APPROVED** [review]

**Pull request merged**